lib/x86simdsort.h

```diff
@@ -34,5 +51,19 @@ template <typename T>
 XSS_EXPORT_SYMBOL std::vector<size_t>
 argselect(T *arr, size_t k, size_t arrsize, bool hasnan = false);
 
+// sort an object
+template <typename T, typename F>
+XSS_EXPORT_SYMBOL void object_qsort(T *arr, size_t arrsize, const F key_func)
+{
+    using return_type_of =
+            typename decltype(std::function {key_func})::result_type;
```

Comment:
This is probably a C++17 technique (CTAD). If you need to support pre-C++17, you may need to rewrite it.